src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1784,55 +1784,70 @@ public bool IsOnTypeHeader(SyntaxNode root, int position)
                 return false;
             }
 
-            var start = GetStartOfNodeExcludingAttributes(typeDecl);
-            var end = typeDecl.GetTypeParameterList()?.GetLastToken().FullSpan.End ??
-                        typeDecl.Identifier.FullSpan.End;
-
-            return position >= start && position <= end;
+            return IsOnHeader(position, typeDecl, typeDecl.Identifier);
         }
 
-        public bool IsInPropertyDeclarationHeader(SyntaxNode node)
+        public bool IsOnPropertyDeclarationHeader(SyntaxNode root, int position, out SyntaxNode propertyDeclaration)
         {
-            var propertyDeclaration = node.GetAncestorOrThis<PropertyDeclarationSyntax>();
+            var node = TryGetAncestorForLocation<PropertyDeclarationSyntax>(position, root);
+            propertyDeclaration = node;
             if (propertyDeclaration == null)
             {
                 return false;
             }
 
-            return IsInHeader(node, propertyDeclaration, propertyDeclaration.Identifier);
+            return IsOnHeader(position, node, node.Identifier);
```

Comment:
nit: `return propertyDeclaration != null && IsOnHeader(...)`

Reply:
Honestly, I'm not a fan of that. Sure, this is longer but feels more obvious.

That said, it would be great if there was return expression so that you could do things like:
`propertyDeclaration ?? return false`

Reply:
not a problem.